### DISCUSSION TITLE
Question about the Math in the dai to borrow for improving understanding. (related to Units)

### DISCUSSION THREAD

**Discussion by WeTheArtist** [smartcontractkit/full-blockchain-solidity-course-py#654] [Q&A] [answered]
So I noticed that in the aave_borrow.py code there is this line:
```ruby
amount_dai_to_borrow = (1 / dai_eth_price) * (borrowable_eth * 0.95)
```
I understand that the answer is correct but I think I am not able to understand some parts of this math so I wanted to check what it is that I am thinking incorrectly.
I was a bit confused with the units (please do correct me where I am wrong)

so, lets start with dai_eth_price
This is the amount of dai in unit eth right?
That means it has dai/eth as units. So at the moment DAI to ETH is 0.00026826 which I think means, that 1 DAI equals 0.00026826 ETH. right?
So DAI/ETH( dai divided by eth) = 0.00026826

now borrowable_eth,
This is the amount of eth that we can borrow in unit eth right?
That means that it has eth as units?. So when I run the code and it shows me : you can borrow 0.0825 worth of ETH! 
it means that I can borrow 0.0825 ETH right? 
So the unit here is 0.0825 ETH right?

So when I try to do the math,
amount_dai_to_borrow = (1 / dai_eth_price) * (borrowable_eth * 0.95) = (1/0.00026826 )*(0.0825 *0.95)
looking at the same math in terms of units
amount_dai_to_borrow = (1 / dai_eth_price) * (borrowable_eth * 0.95) = (1/(DAI/ETH) )*(ETH *0.95)=(ETH/DAI)*(ETH*0.95)

So I am seeing that the unit of ETH does cancel out it becomes (ETH^2)/DAI
I am having difficulty in understanding it.
I was expecting that the ETH units would cancel and DAI would remain, so something in lines of
amount_dai_to_borrow = (dai_eth_price) * (borrowable_eth * 0.95) = (DAI/ETH )*(ETH *0.95)=DAI*0.95

I am very confused about what I am not understanding correctly. Much appreciated if someone can let me know :)

**Comment by ZeroEkkusu** [comment, marked as answer]
```python
amount_dai_to_borrow = (1 / dai_eth_price) * (borrowable_eth * 0.95)
```

For simplicity
- let's say 1 ETH is 1000 DAI
- you have deposited 1 ETH (ignore the interest)
- LTV is 80% and you have not borrowed anything ⇒ `borrowable_eth` is 0.8 ETH

The above code then means:
??? DAI = (1 / 0.001 ETH) * (0.8 ETH * 95%)
??? DAI = 1000 DAI * 0.76 ETH
??? DAI = 760

Which is basically a way of saying:
*If 1 DAI is 0.001 ETH, meaning 1 ETH is 1000 DAI, how much DAI can I borrow with 95% of my 0.8 ETH?
1 : 1000 = 0.76 : ???
??? x 1 = 1000 x 0.76
??? = 760*

**Reply by WeTheArtist** [reply]
Ah yes, I get it now! Thank you, I was confused with the unit of dai_eth_price, it will be in ETH and not in DAI/ETH. Dont know what I was thinking then.

**Reply by ZeroEkkusu** [reply]
Yeah, DAI/ETH is a ratio, while ETH or DAI are units. I updated the answer to match the original code.

**Reply by cromewar** [reply]
Thanks @ZeroEkkusu  you are amazing, and yes that's a perfect explanation.
I'm closing this thread now.
